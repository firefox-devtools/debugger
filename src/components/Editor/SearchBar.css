.search-bar {
  display: flex;
  flex-direction: column;
}

.search-bar .search-field {
  padding-left: 7px;
  height: var(--editor-searchbar-height);
}

.search-bar .close-btn {
  padding: 6px;
}

.search-bottom-bar * {
  -moz-user-select: none;
  user-select: none;
}

.search-bottom-bar {
  display: flex;
  flex-shrink: 0;
  justify-content: flex-end;
  width: calc(100% - 1px);
  height: var(--editor-second-searchbar-height);
  background-color: var(--theme-toolbar-background);
  border-bottom: 1px solid var(--theme-splitter-color);
  padding: 0 13px;
}

.search-bottom-bar button:focus {
  outline: none;
}

.search-bottom-bar .search-modifiers {
  display: flex;
  align-items: center;
}

.search-bottom-bar .search-modifiers button {
  padding: 0 3px;
  margin: 0 3px;
  border: none;
  background: none;
  width: 20px;
  height: 20px;
  border-radius: 3px;
}

.search-bottom-bar .search-modifiers button i {
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0;
  width: 16px;
}

.search-bottom-bar .search-modifiers button svg {
  fill: var(--theme-comment-alt);
  height: 16px;
  width: 16px;
}

.search-bottom-bar .search-modifiers button:hover {
  background: var(--theme-toolbar-background-hover);
}

.search-bottom-bar .search-modifiers button:active {
  outline: none;
}

.search-bottom-bar .search-modifiers button.active svg {
  fill: var(--theme-selection-background);
}

.theme-dark .search-bottom-bar .search-modifiers button.active svg {
  fill: white;
}

.search-bottom-bar .search-type-toggles {
  display: flex;
  align-items: center;
  max-width: 68%;
}

.search-bottom-bar .search-type-name {
  padding: 1px 0 0 0;
  margin: 0 0 0 6px;
  border: none;
  background: transparent;
  color: var(--theme-comment-alt);
}

.search-bottom-bar .search-type-toggles .search-type-btn:active {
  outline: none;
}

.search-bottom-bar .search-type-toggles .search-type-btn.active {
  color: var(--theme-selection-background);
}

.theme-dark .search-bottom-bar .search-type-toggles .search-type-btn.active {
  color: white;
}

.search-bar .result-list {
  max-height: 230px;
}
