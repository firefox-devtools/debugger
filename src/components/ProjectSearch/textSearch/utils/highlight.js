// Maybe reuse file search's functions?
import React from "react";

export function highlightMatches(lineMatch) {
  const { value, column, match } = lineMatch;
  const len = match.length;

  return (
    <span className="line-value">
      <span className="line-match" key={0}>
        {value.slice(0, column)}
      </span>
      <span className="query-match" key={1}>
        {value.substr(column, len)}
      </span>
      <span className="line-match" key={2}>
        {value.slice(column + len, value.length)}
      </span>
    </span>
  );
}
