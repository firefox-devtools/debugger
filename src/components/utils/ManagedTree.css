.tree {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -o-user-select: none;
  user-select: none;

  flex: 1;
  white-space: nowrap;
  overflow: auto;
}

.tree button {
  display: block;
}

.tree .node {
  padding: 2px 5px;
  position: relative;
}

.tree .node.focused {
  color: white;
  background-color: var(--theme-selection-background);
}

html:not([dir="rtl"]) .tree .node > div {
  margin-left: 10px;
}

html[dir="rtl"] .tree .node > div {
  margin-right: 10px;
}

.tree .node.focused svg {
  fill: white;
}

.tree-node button {
  position: fixed;
}
