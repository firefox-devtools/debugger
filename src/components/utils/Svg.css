.worker,
.folder {
    position: relative;
    top: 2px;
}

.arrow,
.folder,
.domain,
.file,
.worker,
.refresh,
.add-button {
    fill: var(--theme-splitter-color);
}

.domain,
.file,
.worker,
.refresh,
.add-button {
    position: relative;
    top: 1px;
}

.domain svg,
.folder svg,
.worker svg,
.refresh svg,
.add-button svg {
    width: 15px;
    fill: currentColor;
}

.file svg {
    width: 13px;
}

html:not([dir="rtl"]) .file svg,
html:not([dir="rtl"]) .domain svg,
html:not([dir="rtl"]) .folder svg,
html:not([dir="rtl"]) .refresh svg,
html:not([dir="rtl"]) .worker svg {
    margin-right: 5px;
}

html[dir="rtl"] .file svg,
html[dir="rtl"] .domain svg,
html[dir="rtl"] .folder svg,
html[dir="rtl"] .worker svg,
html[dir="rtl"] .refresh svg {
    margin-left: 5px;
}
