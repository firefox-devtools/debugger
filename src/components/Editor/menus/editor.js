/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

// @flow

import { bindActionCreators } from "redux";
import { isOriginalId } from "devtools-source-map";

import { copyToTheClipboard } from "../../../utils/clipboard";
import {
  isPretty,
  getRawSourceURL,
  getFilename,
  shouldBlackbox
} from "../../../utils/source";

import { downloadFile } from "../../../utils/utils";

import actions from "../../../actions";

import type { Source, SourceLocation } from "../../../types";

function isMapped(selectedSource) {
  return isOriginalId(selectedSource.id) || !!selectedSource.sourceMapURL;
}

export const continueToHereItem = (
  location: SourceLocation,
  isPaused: boolean,
  editorActions: EditorItemActions
) => ({
  accesskey: L10N.getStr("editor.continueToHere.accesskey"),
  disabled: !isPaused,
  click: () => editorActions.continueToHere(location.line, location.column),
  id: "node-menu-continue-to-here",
  label: L10N.getStr("editor.continueToHere.label")
});

// menu items

const copyToClipboardItem = (
  selectedSource: Source,
  editorActions: EditorItemActions
) => {
  return {
    id: "node-menu-copy-to-clipboard",
    label: L10N.getStr("copyToClipboard.label"),
    accesskey: L10N.getStr("copyToClipboard.accesskey"),
    disabled: false,
    click: () =>
      !selectedSource.isWasm &&
      typeof selectedSource.text == "string" &&
      copyToTheClipboard(selectedSource.text)
  };
};

const copySourceItem = (
  selectedSource: Source,
  selectionText: string,
  editorActions: EditorItemActions
) => {
  if (selectedSource.isWasm) {
    return;
  }

  return {
    id: "node-menu-copy-source",
    label: L10N.getStr("copySource.label"),
    accesskey: L10N.getStr("copySource.accesskey"),
    disabled: selectionText.length === 0,
    click: () => copyToTheClipboard(selectionText)
  };
};

const copySourceUri2Item = (
  selectedSource: Source,
  editorActions: EditorItemActions
) => ({
  id: "node-menu-copy-source-url",
  label: L10N.getStr("copySourceUri2"),
  accesskey: L10N.getStr("copySourceUri2.accesskey"),
  disabled: !selectedSource.url,
  click: () => copyToTheClipboard(getRawSourceURL(selectedSource.url))
});

const jumpToMappedLocationItem = (
  selectedSource: Source,
  location: SourceLocation,
  hasPrettySource: boolean,
  editorActions: EditorItemActions
) => ({
  id: "node-menu-jump",
  label: L10N.getFormatStr(
    "editor.jumpToMappedLocation1",
    isOriginalId(selectedSource.id)
      ? L10N.getStr("generated")
      : L10N.getStr("original")
  ),
  accesskey: L10N.getStr("editor.jumpToMappedLocation1.accesskey"),
  disabled:
    (!isMapped(selectedSource) && !isPretty(selectedSource)) || hasPrettySource,
  click: () => editorActions.jumpToMappedLocation(location)
});

const showSourceMenuItem = (
  selectedSource: Source,
  editorActions: EditorItemActions
) => ({
  id: "node-menu-show-source",
  label: L10N.getStr("sourceTabs.revealInTree"),
  accesskey: L10N.getStr("sourceTabs.revealInTree.accesskey"),
  disabled: !selectedSource.url,
  click: () => editorActions.showSource(selectedSource.id)
});

const blackBoxMenuItem = (
  selectedSource: Source,
  editorActions: EditorItemActions
) => ({
  id: "node-menu-blackbox",
  label: selectedSource.isBlackBoxed
    ? L10N.getStr("sourceFooter.unblackbox")
    : L10N.getStr("sourceFooter.blackbox"),
  accesskey: L10N.getStr("sourceFooter.blackbox.accesskey"),
  disabled: !shouldBlackbox(selectedSource),
  click: () => editorActions.toggleBlackBox(selectedSource)
});

const watchExpressionItem = (
  selectedSource: Source,
  selectionText: string,
  editorActions: EditorItemActions
) => ({
  id: "node-menu-add-watch-expression",
  label: L10N.getStr("expressions.label"),
  accesskey: L10N.getStr("expressions.accesskey"),
  click: () => editorActions.addExpression(selectionText)
});

const evaluateInConsoleItem = (
  selectedSource: Source,
  selectionText: string,
  editorActions: EditorItemActions
) => ({
  id: "node-menu-evaluate-in-console",
  label: L10N.getStr("evaluateInConsole.label"),
  click: () => editorActions.evaluateInConsole(selectionText)
});

const downloadFileItem = (
  selectedSource: Source,
  editorActions: EditorItemActions
) => {
  return {
    id: "node-menu-download-file",
    label: L10N.getStr("downloadFile.label"),
    accesskey: L10N.getStr("downloadFile.accesskey"),
    click: () => downloadFile(selectedSource, getFilename(selectedSource))
  };
};

export function editorMenuItems({
  editorActions,
  selectedSource,
  location,
  selectionText,
  hasPrettySource,
  isTextSelected,
  isPaused
}: {
  editorActions: EditorItemActions,
  selectedSource: Source,
  location: SourceLocation,
  selectionText: string,
  hasPrettySource: boolean,
  isTextSelected: boolean,
  isPaused: boolean
}) {
  const items = [];

  items.push(
    jumpToMappedLocationItem(
      selectedSource,
      location,
      hasPrettySource,
      editorActions
    ),
    continueToHereItem(location, isPaused, editorActions),
    { type: "separator" },
    copyToClipboardItem(selectedSource, editorActions),
    copySourceItem(selectedSource, selectionText, editorActions),
    copySourceUri2Item(selectedSource, editorActions),
    downloadFileItem(selectedSource, editorActions),
    { type: "separator" },
    showSourceMenuItem(selectedSource, editorActions),
    blackBoxMenuItem(selectedSource, editorActions)
  );

  if (isTextSelected) {
    items.push(
      { type: "separator" },
      watchExpressionItem(selectedSource, selectionText, editorActions),
      evaluateInConsoleItem(selectedSource, selectionText, editorActions)
    );
  }

  return items;
}

export type EditorItemActions = {
  addExpression: typeof actions.addExpression,
  continueToHere: typeof actions.continueToHere,
  evaluateInConsole: typeof actions.evaluateInConsole,
  flashLineRange: typeof actions.flashLineRange,
  jumpToMappedLocation: typeof actions.jumpToMappedLocation,
  showSource: typeof actions.showSource,
  toggleBlackBox: typeof actions.toggleBlackBox
};

export function editorItemActions(dispatch: Function) {
  return bindActionCreators(
    {
      addExpression: actions.addExpression,
      continueToHere: actions.continueToHere,
      evaluateInConsole: actions.evaluateInConsole,
      flashLineRange: actions.flashLineRange,
      jumpToMappedLocation: actions.jumpToMappedLocation,
      showSource: actions.showSource,
      toggleBlackBox: actions.toggleBlackBox
    },
    dispatch
  );
}
