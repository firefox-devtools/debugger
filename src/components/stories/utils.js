const renderComponentFromFixture =
      require("../../test/utils/renderComponentFromFixture");

const { storiesOf } = require("@kadira/storybook");

module.exports = {
  renderComponent: renderComponentFromFixture,
  storiesOf
};
