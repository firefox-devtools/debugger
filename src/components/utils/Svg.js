/**
 * This file maps the SVG React Components in the assets/images directory.
 */
const Svg = require("../../../assets/images/Svg");

require("./Svg.css");

module.exports = Svg;
