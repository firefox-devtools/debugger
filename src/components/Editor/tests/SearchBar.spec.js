import React from "react";
import { shallow } from "enzyme";
import SearchBar from "../SearchBar";
import "../../../workers/search";
import "../../../utils/editor";

const SearchBarComponent = SearchBar.WrappedComponent;

jest.mock("../../../workers/search", () => ({
  getMatches: () => Promise.resolve(["result"])
}));

jest.mock("../../../utils/editor", () => ({
  find: () => ({ ch: "1", line: "1" })
}));

function generateDefaults() {
  return {
    query: "",
    searchOn: true,
    symbolSearchOn: true,
    editor: {},
    searchResults: {},
    selectedSymbolType: "functions",
    selectedSource: {
      get: () => " text text query text"
    },
    setFileSearchQuery: msg => msg,
    symbolSearchResults: [],
    modifiers: {
      get: jest.fn(),
      toJS: () => ({ caseSensitive: true, wholeWord: false, regexMatch: false })
    },
    selectedResultIndex: 0,
    updateSearchResults: jest.fn(),
    doSearch: jest.fn()
  };
}

function render(overrides = {}) {
  const defaults = generateDefaults();
  const props = { ...defaults, ...overrides };
  const component = shallow(<SearchBarComponent {...props} />);
  return { component, props };
}

describe("SearchBar", () => {
  it("should render", () => {
    const { component } = render();
    expect(component).toMatchSnapshot();
  });
});

describe("doSearch", () => {
  it("should complete a search", async () => {
    const { component, props } = render();
    component
      .find("SearchInput")
      .simulate("change", { target: { value: "query" } });

    const doSearchArgs = props.doSearch.mock.calls[0][0];
    expect(doSearchArgs).toMatchSnapshot();
  });
});
