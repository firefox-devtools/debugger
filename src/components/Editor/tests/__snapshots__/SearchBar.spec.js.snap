// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar should render 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="search-bar"
>
    <SearchInput
        count={undefined}
        handleClose={[Function]}
        handleNext={[Function]}
        handlePrev={[Function]}
        onChange={[Function]}
        onKeyUp={[Function]}
        placeholder="Search in file…"
        query=""
        showErrorEmoji={true}
        size=""
        summaryMsg=""
    />
    <div
        className="search-bottom-bar"
    >
        <div
            className="search-type-toggles"
        >
            <span
                className="search-type-name"
                onClick={[Function]}
            >
                Search functions…
            </span>
        </div>
        <div
            className="search-modifiers"
        >
            <span
                className="search-type-name"
            >
                Modifiers:
            </span>
            <SearchModBtn
                className="regex-match-btn"
                modVal="regexMatch"
                svgName="regex-match"
                tooltip="Regex"
            />
            <SearchModBtn
                className="case-sensitive-btn"
                modVal="caseSensitive"
                svgName="case-match"
                tooltip="Case sensitive"
            />
            <SearchModBtn
                className="whole-word-btn"
                modVal="wholeWord"
                svgName="whole-word-match"
                tooltip="Whole word"
            />
        </div>
    </div>
</div>,
  "nodes": Array [
    <div
      className="search-bar"
>
      <SearchInput
            count={undefined}
            handleClose={[Function]}
            handleNext={[Function]}
            handlePrev={[Function]}
            onChange={[Function]}
            onKeyUp={[Function]}
            placeholder="Search in file…"
            query=""
            showErrorEmoji={true}
            size=""
            summaryMsg=""
      />
      <div
            className="search-bottom-bar"
      >
            <div
                  className="search-type-toggles"
            >
                  <span
                        className="search-type-name"
                        onClick={[Function]}
                  >
                        Search functions…
                  </span>
            </div>
            <div
                  className="search-modifiers"
            >
                  <span
                        className="search-type-name"
                  >
                        Modifiers:
                  </span>
                  <SearchModBtn
                        className="regex-match-btn"
                        modVal="regexMatch"
                        svgName="regex-match"
                        tooltip="Regex"
                  />
                  <SearchModBtn
                        className="case-sensitive-btn"
                        modVal="caseSensitive"
                        svgName="case-match"
                        tooltip="Case sensitive"
                  />
                  <SearchModBtn
                        className="whole-word-btn"
                        modVal="wholeWord"
                        svgName="whole-word-match"
                        tooltip="Whole word"
                  />
            </div>
      </div>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SearchBar
        editor={Object {}}
        modifiers={
                Object {
                        "get": [Function],
                        "toJS": [Function],
                      }
        }
        query=""
        searchOn={true}
        searchResults={Object {}}
        selectedResultIndex={0}
        selectedSource={
                Object {
                        "get": [Function],
                      }
        }
        selectedSymbolType="functions"
        setFileSearchQuery={[Function]}
        symbolSearchOn={true}
        symbolSearchResults={Array []}
        updateSearchResults={[Function]}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SearchBar {
        "_reactInternalInstance": [Circular],
        "buildSummaryMsg": [Function],
        "clearSearch": [Function],
        "closeSearch": [Function],
        "context": Object {
          "shortcuts": undefined,
        },
        "doSearch": [Function],
        "onChange": [Function],
        "onEscape": [Function],
        "onKeyUp": [Function],
        "props": Object {
          "editor": Object {},
          "modifiers": Object {
            "get": [Function],
            "toJS": [Function],
          },
          "query": "",
          "searchOn": true,
          "searchResults": Object {},
          "selectedResultIndex": 0,
          "selectedSource": Object {
            "get": [Function],
          },
          "selectedSymbolType": "functions",
          "setFileSearchQuery": [Function],
          "symbolSearchOn": true,
          "symbolSearchResults": Array [],
          "updateSearchResults": [Function],
        },
        "refs": Object {},
        "renderSearchModifiers": [Function],
        "searchContents": [Function],
        "searchInput": [Function],
        "selectSearchInput": [Function],
        "setSearchValue": [Function],
        "state": Object {
          "count": 0,
          "index": -1,
          "selectedResultIndex": 0,
        },
        "toggleSearch": [Function],
        "traverseResults": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": Object {
        "_currentElement": <div
          className="search-bar"
>
          <SearchInput
                    count={undefined}
                    handleClose={[Function]}
                    handleNext={[Function]}
                    handlePrev={[Function]}
                    onChange={[Function]}
                    onKeyUp={[Function]}
                    placeholder="Search in file…"
                    query=""
                    showErrorEmoji={true}
                    size=""
                    summaryMsg=""
          />
          <div
                    className="search-bottom-bar"
          >
                    <div
                              className="search-type-toggles"
                    >
                              <span
                                        className="search-type-name"
                                        onClick={[Function]}
                              >
                                        Search functions…
                              </span>
                    </div>
                    <div
                              className="search-modifiers"
                    >
                              <span
                                        className="search-type-name"
                              >
                                        Modifiers:
                              </span>
                              <SearchModBtn
                                        className="regex-match-btn"
                                        modVal="regexMatch"
                                        svgName="regex-match"
                                        tooltip="Regex"
                              />
                              <SearchModBtn
                                        className="case-sensitive-btn"
                                        modVal="caseSensitive"
                                        svgName="case-match"
                                        tooltip="Case sensitive"
                              />
                              <SearchModBtn
                                        className="whole-word-btn"
                                        modVal="wholeWord"
                                        svgName="whole-word-match"
                                        tooltip="Whole word"
                              />
                    </div>
          </div>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="search-bar"
>
          <SearchInput
                    count={undefined}
                    handleClose={[Function]}
                    handleNext={[Function]}
                    handlePrev={[Function]}
                    onChange={[Function]}
                    onKeyUp={[Function]}
                    placeholder="Search in file…"
                    query=""
                    showErrorEmoji={true}
                    size=""
                    summaryMsg=""
          />
          <div
                    className="search-bottom-bar"
          >
                    <div
                              className="search-type-toggles"
                    >
                              <span
                                        className="search-type-name"
                                        onClick={[Function]}
                              >
                                        Search functions…
                              </span>
                    </div>
                    <div
                              className="search-modifiers"
                    >
                              <span
                                        className="search-type-name"
                              >
                                        Modifiers:
                              </span>
                              <SearchModBtn
                                        className="regex-match-btn"
                                        modVal="regexMatch"
                                        svgName="regex-match"
                                        tooltip="Regex"
                              />
                              <SearchModBtn
                                        className="case-sensitive-btn"
                                        modVal="caseSensitive"
                                        svgName="case-match"
                                        tooltip="Case sensitive"
                              />
                              <SearchModBtn
                                        className="whole-word-btn"
                                        modVal="wholeWord"
                                        svgName="whole-word-match"
                                        tooltip="Whole word"
                              />
                    </div>
          </div>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SearchBar
    editor={Object {}}
    modifiers={
        Object {
            "get": [Function],
            "toJS": [Function],
          }
    }
    query=""
    searchOn={true}
    searchResults={Object {}}
    selectedResultIndex={0}
    selectedSource={
        Object {
            "get": [Function],
          }
    }
    selectedSymbolType="functions"
    setFileSearchQuery={[Function]}
    symbolSearchOn={true}
    symbolSearchResults={Array []}
    updateSearchResults={[Function]}
/>,
}
`;

exports[`doSearch should complete a search 1`] = `
Object {
  "count": 1,
  "index": "1",
  "matchIndex": -1,
  "matches": Array [
    "result",
  ],
}
`;
